### PR TITLE
Dealii dev

### PR DESCRIPTION
**Pull request by Rombur**
Update the docker image and mfmg to use the development version of deal. The reason for this change is that it fixes the incompatibility between deal.II and trilinos. Right now, you need to be lucky when you pick a version of Trilinos. This has been "fixed" in deal by making sacado an optional dependency.

This also fixes the bug found by @dalg24 when using `mmult`.

 Note that you will need to update your version of deal.II

### CODE REVIEW

File: ci/Dockerfile [hunk 1]
```diff
@@ -1,6 +1,6 @@
 FROM nvidia/cuda:8.0-devel
 
-ENV N_PROCS=4
+ENV N_PROCS=1
```
Comment (dalg24) [inline ci/Dockerfile]: Why did you change the default?

File: ci/Dockerfile [hunk 2]
```diff
@@ -36,7 +36,7 @@ RUN mkdir -p ${PREFIX} && \
     mkdir source && \
     mkdir build
 
-RUN export CMAKE_VERSION=3.11.1 && \
+RUN export CMAKE_VERSION=3.11.4 && \
```
Comment (dalg24) [inline ci/Dockerfile]: What version of CMake do we actually require?  Just curious.

File: ci/Dockerfile (hunk 1 above)
Comment (Rombur) [reply ci/Dockerfile]: Because I can't compile with 4 procs on my desktop. deal uses all the memory.
